src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,202 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Parser class allows the Duke program
+ * to parse user inputs into commands.
+ *
+ * @author Gerald Teo Jin Wei
+ * @version 0.1
+ * @since 2022-08-28
+ */
+public class Parser {
+    public Parser() {}
+
+    /**
+     * This method is used to convert the date in YYYY-MM-DD format to MMM d YYYY
+     * (e.g 2019-10-02 to OCT 2 2019)
+     * @param date This is string of date in YYYY-MM-DD format
+     * @return String This returns the string of the date in the MMM d YYYY format
+     */
+    public static String parseDate(String date) {
+        LocalDate d = LocalDate.parse(date);
+        String formattedDate = d.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+        return formattedDate;
+    }
+
+    /**
+     * This method is used to convert the user's string input into commands like
+     * quit, list out tasks, adding/removing/marking/finding specific tasks. When
+     * an invalid input is received, an error is thrown and user can try another
+     * input
+     * @param command This is the user's string input
+     * @param taskList This is the user's current list of tasks
+     * @param ui This is for the Duke program to interact/print outputs to the user
+     */
+    public static void parseCommand(String command, TaskList taskList, Ui ui) {
+        try {
+            if (command.equals("bye")) {
+                ui.quit();
+            } else if (command.equals("list")) {
+                ui.listOutTasks(taskList);
+            } else {
+                String[] splitStr = command.split(" ");
+                Task taskToMark = getTaskToMark(splitStr, taskList);
+                Task taskToAdd = getTaskToAdd(command);
+                Task taskToDelete = getTaskToDelete(splitStr, taskList);
+                List<Task> taskListWithKeyWord = getAllTaskWithKeyword(command, taskList);
+                if (taskToMark != null) {
+                    String action = splitStr[0];
+                    if (action.equals("mark")) {
+                        ui.markTask(taskToMark);
+                    } else {
+                        ui.unmarkTask(taskToMark);
+                    }
+                } else if (taskToAdd != null) {
+                    ui.addTask(taskList, taskToAdd);
+                } else if (taskToDelete != null) {
+                    ui.deleteTask(taskList, taskToDelete);
+                } else if (taskListWithKeyWord.size() != 0) {
+                    ui.printTasksWithKeyword(taskListWithKeyWord);
+                } else {
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(\n");
+                }
+            }

```

Comment:
The indentation in this area may be reduced through the use of guard clauses, i.e.
```java
if (command.equals(bye) {
    ui.quit();
    return;
}
if (command.equals(list) {
    ...
    return;
}
String[] splitStr = ...
...
if (taskToMark != null) {
    ...
    return;
}
if (taskToAdd != null) {
    ...
    return;
}
...
throw new DukeException("...");
```